std/assembly/internal/typedarray.ts

```diff
@@ -231,3 +232,47 @@ export function EVERY<TArray extends TypedArray<T>, T>(
   }
   return true;
 }
+
+@inline
+export function FOR_EACH<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => void,
+): void {
+  var buffer = array.buffer;
+  var length = array.length;
+  var byteOffset = array.byteOffset;
+
+  for (let i = 0; i < length; i++) {
+    callbackfn(LOAD<T>(buffer, i, byteOffset), i, array);
+  }
+}
+
+@inline
+export function FILTER<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => bool,
+): TArray {
+  var backingBuffer = new ArrayBuffer(sizeof<T>());
+  var count = 0;
+  var buffer = array.buffer;
+  var length = array.length;
+  var byteOffset = array.byteOffset;
+  var backByteLength = backingBuffer.byteLength;
+  var value: T;
+  for (let i = 0; i < length; i++) {
+    value = LOAD<T>(buffer, i, byteOffset);
+    if (callbackfn(value, i, array)) {
+      backByteLength = backingBuffer.byteLength;
+      if (backByteLength < (count + 1) << alignof<T>()) {
+        backingBuffer = reallocateUnsafe(backingBuffer, backByteLength << 1);
```

Comment:
Not related to your PR directly: One thing that bothers me about reallocateUnsafe vs allocateUnsafe is that the former zeroes memory whenever newSize > oldSize, while the latter doesn't bother with zeroing anything. Somewhat inconsistent and thus likely to change, in turn making it necessary to review any code that uses it.

Reply:
```ts
```ts
export function growUnsafe(buffer: ArrayBuffer, newByteLength: i32): ArrayBuffer {
  var oldByteLength = buffer.byteLength;
  assert(oldByteLength < newByteLength && newByteLength < MAX_BLENGTH);
  if (newByteLength <= <i32>(computeSize(oldByteLength) - HEADER_SIZE)) {
    store<i32>(changetype<usize>(buffer), newByteLength, offsetof<ArrayBuffer>("byteLength"));
  } else {
    let newBuffer = allocateUnsafe(newByteLength);
    memory.copy(
      changetype<usize>(newBuffer) + HEADER_SIZE,
      changetype<usize>(buffer) + HEADER_SIZE,
      <usize>oldByteLength
    );
    if (!isManaged<ArrayBuffer>()) {
      memory.free(changetype<usize>(buffer));
    }
    buffer = newBuffer;
  }
  return buffer;
}
```

Reply:
Sorry for relay. But `growUnsafe` is unnecessary because we already have `reallocateUnsafe` which do the same

Reply:
Alright. I switch my strategy to this.